src/Confluent.Kafka/Headers.cs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     A collection of Kafka message headers.
+    /// </summary>
+    /// <remarks>
+    ///     Message headers are supported by v0.11 brokers and above.
+    /// </remarks>
+    public class Headers : IEnumerable<Header>
+    {
+        private readonly List<Header> headers = new List<Header>();
+
+        /// <summary>
+        ///     Append a new header to the collection.
+        /// </summary>
+        /// <param name="key">
+        ///     The header key.
+        /// </param>
+        /// <param name="val">
+        ///     The header value (possibly null). Note: A null
+        ///     header value is distinct from an empty header
```

Comment:
ah!

Reply:
never believe comments ... (this is fixed now, with test).